src/pykeen/sampling/basic_negative_sampler.py

```diff
@@ -2,60 +2,122 @@
 
 """Negative sampling algorithm based on the work of of Bordes *et al.*."""
 
+from typing import Collection, Optional, Tuple
+
 import torch
 
 from .negative_sampler import NegativeSampler
+from ..triples import TriplesFactory
 
 __all__ = [
     'BasicNegativeSampler',
 ]
 
+LOOKUP = {'h': 0, 'r': 1, 't': 2}
+
 
 class BasicNegativeSampler(NegativeSampler):
     r"""A basic negative sampler.
 
-    This negative sampler that corrupts positive triples $(h,r,t) \in \mathcal{K}$ by replacing either $h$ or $t$.
+    This negative sampler that corrupts positive triples $(h,r,t) \in \mathcal{K}$ by replacing either $h$, $r$ or $t$
```

Comment:
@cthoyt do we need to have `:math:` before Latex code?

Reply:
Nope there's a Sphinx extension for using inline $